src/main/java/org/apache/sysml/runtime/instructions/cp/ParamservBuiltinCPInstruction.java

```diff
@@ -381,26 +405,17 @@ private void doDataPartition(ExecutionContext ec, List<LocalPSWorker> workers) {
 				throw new DMLRuntimeException(String.format("Paramserv function: not support data partition scheme '%s'", getParam(PS_SCHEME)));
 			}
 		}
-		switch (scheme) {
-		case DISJOINT_CONTIGUOUS:
-			disjointContiguous(features, labels, valFeatures, valLabels, workers);
-			break;
-		case DISJOINT_RANDOM:
-		case OVERLAP_RESHUFFLE:
-		case DISJOINT_ROUND_ROBIN:
-			throw new DMLRuntimeException(String.format("Paramserv function: the scheme '%s' is not supported.", scheme));
-		}
+		return scheme;
 	}
 
-	private void disjointContiguous(MatrixObject features, MatrixObject labels, MatrixObject valFeatures,
+	private void doDataPartition(DataPartitioner dp, MatrixObject features, MatrixObject labels, MatrixObject valFeatures,
 			MatrixObject valLabels, List<LocalPSWorker> workers) {
-		// training data
-		List<MatrixObject> pfs = disjointContiguous(workers.size(), features);
-		List<MatrixObject> pls = disjointContiguous(workers.size(), labels);
+		List<MatrixObject> pfs = dp.doPartition(workers.size(), features);
+		List<MatrixObject> pls = dp.doPartition(workers.size(), labels);
```

Comment:
This separate partitioning worked well for disjoint contiguous because it is a deterministic partitioning scheme. However, for random reshuffling (disjoint random and overlap reshuffle) this creates two different permutation matrices and thus mixes observations and labels arbitrarily. In general, we would need to reuse the same permutation for features and labels. Hence, I would recommend to extend the abstraction of `DataPartitioner` accordingly.